osu.Game/Skinning/SkinModelManager.cs

```diff
@@ -49,8 +51,36 @@ public SkinModelManager(Storage storage, RealmContextFactory contextFactory, Gam
 
         protected override Task Populate(SkinInfo model, ArchiveReader? archive, Realm realm, CancellationToken cancellationToken = default)
         {
-            if (string.IsNullOrEmpty(model.InstantiationInfo))
-                model.InstantiationInfo = createInstance(model).GetType().GetInvariantInstantiationInfo();
+            var skinInfoFile = model.Files.SingleOrDefault(f => f.Filename == skin_info_file);
+
+            if (skinInfoFile != null)
+            {
+                try
+                {
+                    using (var existingStream = Files.Storage.GetStream(skinInfoFile.File.GetStoragePath()))
+                    using (var reader = new StreamReader(existingStream))
+                    {
+                        var deserialisedSkinInfo = JsonConvert.DeserializeObject<SkinInfo>(reader.ReadToEnd());
```

Comment:
While this works, I'd say that maybe it's dumping a bit too much info. Here's an example to show what I mean:

```json
{
  "Name": "osu! (triangles) (modified)",
  "Creator": "Unknown",
  "Hash": "474e3ae1925613810d7c49b1c8bed2946ef21fd6d6f999f66a9939b219f44388", // probably shouldn't be here
  "Protected": false, // probably shouldn't be here
  "InstantiationInfo": "osu.Game.Skinning.DefaultSkin, osu.Game",
  "Files": [ // this entire property is a little redundant
    {
      "File": {
        "Hash": "0108d0eb129aea08415719722873bc29805ada109479a33b78fe8ff1e82a4008" // probably shouldn't be here
      },
      "Filename": "skin.ini",
      "Parent": null // inherited from EmbeddedObject, probably shouldn't be here
    },
    {
      "File": {
        "Hash": "c7bc3465f74ffdfc545f999b5cb4f64a545da90cdccccdceca963fb8c2c8e74a"
      },
      "Filename": "skininfo.json",
      "Parent": null
    },
    {
      "File": {
        "Hash": "31cd6aa52bd39ec58d874bba33857781bf6d392a5a0b6e7feba0e7976f94e368"
      },
      "Filename": "MainHUDComponents.json",
      "Parent": null
    }
  ],
  "DeletePending": false // probably shouldn't be here
}
```

Reply:
Yeah, I have a TODO which is to go through each serialisable model and make them explicit include only for json, but it does make sense to fix this one here while we're at it I guess.

Reply:
I've made this opt-in now, looks like this:

```json
{
  "Name": "osu! (triangles) (modified)",
  "Creator": "Unknown",
  "InstantiationInfo": "osu.Game.Skinning.DefaultSkin, osu.Game"
}
```